File: scripts/generate_share_url.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

import os
import sys
import requests
import json

# 基础URL
BASE_URL = os.environ.get("AIBOX_BASE_URL", "http://aidev.nicebox.cn/api/openclaw")

# API路径
ENDPOINT_SHARE_URL = "/site/generateShareUrl"

def load_config():
    """加载配置"""
    # 从环境变量构造
    base_url = os.environ.get("AIBOX_BASE_URL", "http://aidev.nicebox.cn/api/openclaw")
    api_key = os.environ.get("AIBOX_API_KEY", "")
    
    if not api_key:
        print("错误：缺少 API 配置，请设置 AIBOX_API_KEY 环境变量")
        sys.exit(1)
    
    return {
        "api_url": base_url,
        "api_key": api_key
    }

def generate_share_url(config):
    """生成临时分享地址"""
    url = f"{config['api_url']}{ENDPOINT_SHARE_URL}"
    headers = {
        "Authorization": config['api_key'],
        "Content-Type": "application/json"
    }
    
    try:
        response = requests.get(url, headers=headers)
        result = response.json()
        
        if result.get("code") == 0:
            data = result.get("data", {})
            share_url = data.get("share_url", "")
            print("\n✅ 临时分享链接生成成功：")
            print(f"   分享链接: {share_url}")
            print(f"   有效期:   2 小时\n")
        else:
            print(f"错误：{result.get('message', '生成分享地址失败')}")
            sys.exit(1)
            
    except Exception as e:
        print(f"错误：{str(e)}")
        sys.exit(1)

def main():
    """主函数"""
    config = load_config()
    generate_share_url(config)

if __name__ == "__main__":
    main()

File: scripts/generate_share_url.mjs
#!/usr/bin/env node

import fetch from 'node-fetch';

const BASE_URL        = process.env.AIBOX_BASE_URL || "https://ai.nicebox.cn/api/openclaw";
const ENDPOINT_LANG  = "/site_pages/getLanguageList";
const ENDPOINT_SHARE = "/site/generateShareUrl";

function loadConfig() {
  const apiKey = process.env.AIBOX_API_KEY || "";
  if (!apiKey) {
    console.error('错误：缺少 API 配置，请设置 AIBOX_API_KEY 环境变量');
    process.exit(1);
  }
  return { api_url: BASE_URL, api_key: apiKey };
}

/**
 * 检查站点语言配置
 * total > 0 → 有内容 → 继续生成
 * total = 0 → 无内容 → 终止操作
 */
async function checkSiteHasContent(config) {
  const resp = await fetch(config.api_url + ENDPOINT_LANG, {
    method: 'GET',
    headers: { "Authorization": config.api_key }
  });
  const result = await resp.json();

  if (result.code !== 0 || !result.data) {
    console.error(`错误：无法获取站点语言配置 — ${result.message || '未知错误'}`);
    process.exit(1);
  }

  const total = result.data.total ?? 0;
  const list  = result.data.list   ?? [];

  if (total === 0) {
    console.error('\n[操作终止] 站点暂无网站内容，请先生成并发布网站后再生成分享链接。');
    process.exit(1);
  }

  console.log(`✓ 站点内容检测通过（语言配置 ${total} 项）`);
  return total;
}

async function generateShareUrl(config) {
  const resp = await fetch(config.api_url + ENDPOINT_SHARE, {
    method: 'GET',
    headers: { "Authorization": config.api_key }
  });
  const result = await resp.json();

  if (result.code === 0 && result.data) {
    console.log('\n✅ 临时分享链接生成成功：');
    console.log('   分享链接: ' + result.data.share_url);
    console.log('   有效期:   2 小时\n');
  } else {
    console.error(`错误：${result.message || '生成分享地址失败'}`);
    process.exit(1);
  }
}

async function main() {
  const config = loadConfig();
  await checkSiteHasContent(config);
  await generateShareUrl(config);
}

main();
